src/Avalonia.Base/Styling/Activators/NthChildActivator.cs

```diff
@@ -28,24 +30,50 @@ internal sealed class NthChildActivator : StyleActivatorBase
 
         protected override bool EvaluateIsActive()
         {
-            return NthChildSelector.Evaluate(_control, _provider, _step, _offset, _reversed).IsMatch;
+            var index = _index >= 0 ? _index : _provider.GetChildIndex(_control);
+            return NthChildSelector.Evaluate(index, _provider, _step, _offset, _reversed).IsMatch;
         }
 
-        protected override void Initialize() => _provider.ChildIndexChanged += ChildIndexChanged;
-        protected override void Deinitialize() => _provider.ChildIndexChanged -= ChildIndexChanged;
+        protected override void Initialize()
+        {
+            _provider.ChildIndexChanged += ChildIndexChanged;
+            _provider.TotalCountChanged += TotalCountChanged;
+        }
+
+        protected override void Deinitialize()
+        {
+            _provider.ChildIndexChanged -= ChildIndexChanged;
```

Comment:
1. Do we even need to subscribe on TotalCountChanged when it's not reversed?
2. TotalCountChanged  is not unsubscribed on Deinitialize

Reply:
1. I think this depends on the answer to the question in the PR description
2. This too, really..,